nf_core/components/components_utils.py

```diff
@@ -176,16 +177,70 @@ def get_biotools_id(tool_name) -> str:
         response.raise_for_status()  # Raise an error for bad status codes
         # Parse the JSON response
         data = response.json()
+        return data
 
-        # Iterate through the tools in the response to find the tool name
-        for tool in data["list"]:
-            if tool["name"].lower() == tool_name:
-                return tool["biotoolsCURIE"]
+    except requests.exceptions.RequestException as e:
+        log.warning(f"Could not find bio.tools information for '{tool_name}': {e}")
+        return None
 
-        # If the tool name was not found in the response
-        log.warning(f"Could not find a bio.tools ID for '{tool_name}'")
-        return ""
 
-    except requests.exceptions.RequestException as e:
-        log.warning(f"Could not find a bio.tools ID for '{tool_name}': {e}")
-        return ""
+def get_biotools_id(data: dict, tool_name: str) -> str:
+    """
+    Try to find a bio.tools ID for 'tool'
+    """
+    # Iterate through the tools in the response to find the tool name
+    for tool in data["list"]:
+        if tool["name"].lower() == tool_name:
+            return tool["biotoolsCURIE"]
+
+    # If the tool name was not found in the response
+    log.warning(f"Could not find a bio.tools ID for '{tool_name}'")
+    return ""
```

Comment:
not sure if it isn't better to always return `None` if we don't get a correct entry, like we do in the function above

Reply:
I had another look and since this is the value we use for the `meta.yml` it is easier to return an empty string that we can directly add to the file. That way, we avoid having to check if it's `None`.